### PR TITLE
Removed `InputModule` from `Choices`

### PR DESCRIPTION
**Pull request by B-rando1**
We had discussed a few times that there isn't much need to put the different pieces related to input (parsing, validation, and storage) in different modules.  Since the `Modular Separated` pattern is not very useful, and in the case of `Bundled` inputs uses some patterns that are not good form (creating an empty object, then populating its members externally), I suggested in #3781 that we remove `InputModule` altogether and assume that all the input functionality is in the same module.

I don't believe we ever got around to discussing this, so we might have to make sure we want to do this.  I talked with @balacij and he agrees that this is a good change to be making.

Notable changes:
- Removed `InputModule` from `Choices` in [Choices.hs](https://github.com/JacquesCarette/Drasil/blob/master/code/drasil-code/lib/Language/Drasil/Choices.hs)
- Renamed `genInputModCombined` to `genInputMod` in [Modules.hs](https://github.com/JacquesCarette/Drasil/blob/master/code/drasil-code/lib/Language/Drasil/Code/Imperative/Modules.hs)
- Changed label of modularity from `'C'` for `Combined` and `'S'` for `Separated` to `'M'` for `Modular` in a few places, e.g. in [Choices.hs](https://github.com/JacquesCarette/Drasil/blob/master/code/drasil-example/projectile/lib/Drasil/Projectile/Choices.hs) from Projectile.
    - This last change caused a large number of changes in `stable`.
- Changed any of the examples which previously used `Modular Separated` to use `Modular`.

Because of the large number of things this change affects I think it's good that we be thorough in double-checking the changes, but if we can reach a consensus soon on whether these are good changes to be making, that would help me with the rest of the changes to `Choices` going forward.

Contributes to #3781

### CODE REVIEW

File: code/drasil-code/lib/Language/Drasil/Code/Imperative/Parameters.hs [hunk 1]
```diff
@@ -45,17 +44,12 @@ getInConstructorParams = do
...
 getInputFormatIns :: GenState [CodeVarChunk]
 getInputFormatIns = do
-  g <- get
-  let getIns :: Structure -> InputModule -> [CodeVarChunk]
-      getIns Bundled Separated = [quantvar inParams]
-      getIns _ _ = []
-  getParams "get_input" In $ quantvar inFileName : getIns (inStruct g) (inMod g)
+  let getIns :: [CodeVarChunk]
+      getIns = []
+  getParams "get_input" In $ quantvar inFileName : getIns
```
Comment (samm82) [inline code/drasil-code/lib/Language/Drasil/Code/Imperative/Parameters.hs:52]: Is there a reason we can't do this? Or this is "scaffolding" for a later change?
```suggestion
  getParams "get_input" In $ quantvar inFileName : []
```

Comment (B-rando1) [reply code/drasil-code/lib/Language/Drasil/Code/Imperative/Parameters.hs:52]: Oh true, that's from the fix I made a minute ago.  I tried simplifying that more, but Haskell couldn't figure out its type.  There might be a better way to do it than where I left it, though.  I'll take a look.

Comment (samm82) [reply code/drasil-code/lib/Language/Drasil/Code/Imperative/Parameters.hs:52]: I think you can do `([] :: [CodeVarChunk])` to specify the type of an empty list

Comment (B-rando1) [reply code/drasil-code/lib/Language/Drasil/Code/Imperative/Parameters.hs]: Turns out I was just messing up the syntax, and we don't even need the `[]`!

**Review by JacquesCarette — APPROVED** [review]

**Review by balacij — APPROVED** [review]

**Review by smiths — APPROVED** [review]
The changes make sense to me.  I looked through the generated code in stable.  I didn't run the code to verify that it works.  It is too bad that we can't currently run test cases on our generated code.  We've talked about doing this before.  @samm82 is getting us in the right direction to do this in the future.

**Comment by JacquesCarette** [comment]
Now there's a conflict that needs fixed before it can be merged.

**Comment by B-rando1** [comment]
This branch was merged as a part of #3642.

**Pull request closed without merging**